### ISSUE TITLE
trouble in cache model?

### ISSUE THREAD

**Issue by morder** [Raizlabs/DBFlow#326]
I just updated to latest version 2.2.1 from 2.1.0 and now I see that error
```
 Caused by: java.lang.IllegalArgumentException: You cannot call this method for a table that has no caching id. Eitheruse one Primary Key or call convertToList()
            at com.raizlabs.android.dbflow.sql.SqlUtils.convertToCacheableList(SqlUtils.java:78)
            at com.raizlabs.android.dbflow.sql.SqlUtils.convertToCacheableList(SqlUtils.java:116)
            at com.raizlabs.android.dbflow.sql.SqlUtils.queryList(SqlUtils.java:54)
            at com.raizlabs.android.dbflow.sql.language.BaseModelQueriable.queryList(BaseModelQueriable.java:35)
            at com.raizlabs.android.dbflow.sql.language.Where.queryList(Where.java:347)
            at com.xxx.app.db.EventsTypesTable.getId(EventsTypesTable.java:53)
            at com.xxx.app.MainActivity2.onCreate(MainActivity2.java:88)
            at android.app.Activity.performCreate(Activity.java:5990)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
            at android.app.ActivityThread.access$800(ActivityThread.java:151)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

What I need to do to solve this problem?

My table

```
public class EventsTable extends BaseCacheableModel {
@Column
    @PrimaryKey(autoincrement = true)
    long id;
...
}
```

**Comment by morder** [comment]
query is
```
List<EventsTypesTable> result = new Select()
                .from(EventsTypesTable.class)
                .where(Condition.column(EventsTypesTable$Table.NAME).is(name))
                .queryList();
```

**Comment by ydanneg** [comment]
I confirm issue afetr making my model inherit from BaseCacheableModel

A single primary key:
```
@Column
@PrimaryKey(autoincrement = true)
long id;
```

**Comment by agrosner** [comment]
@morder @ydanneg can you send me the full models of your classes? i think I know what it is but need more info.

**Comment by morder** [comment]
@agrosner 
```
@Table(databaseName = DxxxxDatabase.NAME,
        uniqueColumnGroups = {@UniqueGroup(groupNumber = 1, uniqueConflict = ConflictAction.IGNORE)})
public class EventsTable extends BaseCacheableModel {

    public static class Data{
        public String name;
        public String description;
        public long date;
        public boolean isPrivate;
        public boolean repeatable;
    }

    @Column
    @PrimaryKey(autoincrement = true)
    long id;

    @Column
    @Unique(unique = false, uniqueGroups = 1)
    int groupId = -1;

    @Column
    @ForeignKey(
            references = {@ForeignKeyReference(columnName = "typeId",
                    columnType = Long.class,
                    foreignColumnName = "id")},
            saveForeignKeyModel = false)
    @Unique(unique = false, uniqueGroups = 1)
    public EventsTypesTable typeId;

    @Column
    public String description;

    @Column
    @Unique(unique = false, uniqueGroups = 1)
    public long date;

    @Column
    public boolean isPrivate;

    @Column
    public boolean repeatable; // every year

    private static final Object mSync = new Object();

    public long getId(){
        return id;
    }

    public static int insertOwn(String name, String description, long date, boolean isPrivate, boolean repeatable){
        return insert(0, name, description, date, isPrivate, repeatable, true);
    }

    public static int insert(int groupId, String name, String description, long date, boolean isPrivate, boolean repeatable, boolean force){
        synchronized (mSync) {
            if (!force && groupId < 0) {
                EventsTable row = new Select().from(EventsTable.class).where().orderBy(false, EventsTable$Table.GROUPID).querySingle();
                if (row == null) {
                    groupId = 1;
                } else {
                    groupId = row.groupId + 1;
                }
            }
            EventsTable row = new EventsTable();
            row.groupId = groupId;
            row.typeId = EventsTypesTable.getRow(name);
            row.description = description;
            row.date = date;
            row.isPrivate = isPrivate;
            row.repeatable = repeatable;
            row.save();
            return groupId;
        }
    }

    public static void update(long eventId, String name, String description, long date, boolean isPrivate, boolean repeatable){
        long typeId = EventsTypesTable.getId(name);
        new Update<>(EventsTable.class)
                .set(
                        Condition.column(EventsTable$Table.TYPEID_TYPEID).eq(typeId),
                        Condition.column(EventsTable$Table.DESCRIPTION).eq(description),
                        Condition.column(EventsTable$Table.DATE).eq(date),
                        Condition.column(EventsTable$Table.ISPRIVATE).eq(isPrivate),
                        Condition.column(EventsTable$Table.REPEATABLE).eq(repeatable)
                ).where(Condition.column(EventsTable$Table.ID).eq(eventId))
                .queryClose();
    }

    public static List<EventsTable> getEvents(int groupId, @Nullable Boolean showPrivate){
        if (showPrivate == null) {
            return new Select().from(EventsTable.class)
                    .where(Condition.column(EventsTable$Table.GROUPID).is(groupId))
                    .queryList();
        } else {
            return new Select().from(EventsTable.class) // error line 121
                    .where(Condition.column(EventsTable$Table.GROUPID).is(groupId))
                    .and(Condition.column(EventsTable$Table.ISPRIVATE).is(showPrivate))
                    .queryList();
        }
    }

    public static EventsTable getEvent(long eventId){
        return new Select().from(EventsTable.class)
                .where(Condition.column(EventsTable$Table.ID).is(eventId))
                .querySingle();
    }

    public static void removeEvent(int groupId, long eventId){
        new Delete()
                .from(EventsTable.class)
                .where(Condition.column(EventsTable$Table.GROUPID).is(groupId))
                .and(Condition.column(EventsTable$Table.ID).is(eventId))
                .queryClose();
    }
}
```

error
```
 java.lang.RuntimeException: java.lang.IllegalArgumentException: You cannot call this method for a table that has no caching id. Eitheruse one Primary Key or call convertToList()
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.raizlabs.android.dbflow.runtime.DBTransactionQueue.run(DBTransactionQueue.java:75)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:  Caused by: java.lang.IllegalArgumentException: You cannot call this method for a table that has no caching id. Eitheruse one Primary Key or call convertToList()
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.raizlabs.android.dbflow.sql.SqlUtils.convertToCacheableList(SqlUtils.java:78)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.raizlabs.android.dbflow.sql.SqlUtils.convertToCacheableList(SqlUtils.java:116)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.raizlabs.android.dbflow.sql.SqlUtils.queryList(SqlUtils.java:54)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.raizlabs.android.dbflow.sql.language.BaseModelQueriable.queryList(BaseModelQueriable.java:35)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.raizlabs.android.dbflow.sql.language.Where.queryList(Where.java:347)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.d.app.db.EventsTable.getEvents(EventsTable.java:121)
09-17 11:30:47.584 21829-22010/com.d.app E/AndroidRuntime:     at com.d.app.db.ContactsTable.getEvents(ContactsTable.java:360)
```

**Comment by glureau** [comment]
The caching system is not working when the primary key is also autoincrement (still the case on the current develop branch).
I wrote a post about it but they asked me to wait the 3.0 version (I hope it will manage this case). (Also made a partial fix for this on my fork, if you need this feature on short term.)

**Comment by simplysoft** [comment]
Are there any updates regarding the support of autoincrement in version 3.0?

**Comment by agrosner** [comment]
verified fixed in 3.0/develop.
